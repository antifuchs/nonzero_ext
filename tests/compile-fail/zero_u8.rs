#[macro_use] extern crate nonzero_ext;

use std::num::NonZeroU8;

fn main() {
    let _a: NonZeroU8 = nonzero!(0u8); //~ ERROR could not evaluate repeat length
    //                  ^ should complain about the zero-ness of the argument    
}

