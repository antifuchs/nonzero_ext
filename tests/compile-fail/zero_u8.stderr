error[E0080]: evaluation of constant value failed
 --> $DIR/zero_u8.rs:8:25
  |
8 |     let _a: NonZeroU8 = nonzero!(0u8);
  |                         ^^^^^^^^^^^^^ attempt to subtract with overflow
  |
  = note: this error originates in a macro outside of the current crate (in Nightly builds, run with -Z external-macro-backtrace for more info)
